src/ray/common/ray_object.cc

```diff
@@ -33,17 +89,22 @@ std::shared_ptr<LocalMemoryBuffer> MakeErrorMetadataBuffer(rpc::ErrorType error_
   return MakeBufferFromString(meta);
 }
 
-RayObject::RayObject(rpc::ErrorType error_type)
-    : RayObject(nullptr, MakeErrorMetadataBuffer(error_type), {}) {}
-
-RayObject::RayObject(rpc::ErrorType error_type, const std::string &append_data)
-    : RayObject(MakeBufferFromString(append_data), MakeErrorMetadataBuffer(error_type),
-                {}) {}
+RayObject::RayObject(rpc::ErrorType error_type, const rpc::RayErrorInfo *ray_error_info) {
+  if (ray_error_info == nullptr) {
+    Init(nullptr, MakeErrorMetadataBuffer(error_type), {});
+    return;
+  }
 
-RayObject::RayObject(rpc::ErrorType error_type, const uint8_t *append_data,
-                     size_t append_data_size)
-    : RayObject(MakeBufferFromString(append_data, append_data_size),
-                MakeErrorMetadataBuffer(error_type), {}) {}
+  RAY_CHECK(ray_error_info->has_actor_init_failure());
```

Comment:
This will be removed in the next PR, but it is here in the process of refactoring